### PR TITLE
Add interfaces to index.d.ts

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add interfaces to` index.d.ts`

Note: Integration tests will be added in an upcoming PR

### CODE REVIEW

File: test/unit/firebase-app.spec.ts [hunk 1]
```diff
@@ -496,7 +497,7 @@ describe('FirebaseApp', () => {
       expect(gcsNamespace).not.be.null;
     });
 
-    it('should return a cached version of Messaging on subsequent calls', () => {
+    it('should return a cached version of Storage on subsequent calls', () => {
```
Comment (lahirumaramba) [inline test/unit/firebase-app.spec.ts]: Fixed a typo. I can move this to a separate PR if we prefer that :)

Comment (hiranya911) [reply test/unit/firebase-app.spec.ts]: Thanks 👍

File: src/index.d.ts [hunk 2]
```diff
@@ -5611,6 +5643,155 @@ declare namespace admin.projectManagement {
...
+  interface RemoteConfigParameter {
+
+    /**
+     * The value to set the parameter to, when none of the named conditions evaluate to `true`.
+     */
+    defaultValue?: RemoteConfigParameterValue;
+
+    /**
+     * A `(condition name, value)` map. The `condition_name` of the highest priority
+     * (the one listed first in the `RemoteConfig`'s conditions list) determines the value of
```
[the remark this answers is not in the document]
Comment (lahirumaramba) [reply src/index.d.ts]: Good catch. Changed the wording to ```The condition name of the highest priority (the one listed first in the Remote Config template's conditions list) determines the value of this parameter.```

File: src/index.d.ts [hunk 3]
```diff
@@ -5611,6 +5643,155 @@ declare namespace admin.projectManagement {
...
+     *
+     * @param template The Remote Config template to be validated.
+     * @returns A promise that fulfills with the validated `RemoteConfigTemplate`.
+     */
+    validateTemplate(template: RemoteConfigTemplate): Promise<RemoteConfigTemplate>;
+
+    /**
+     * Publishes a Remote Config template.
+     *
+     * @param template The Remote Config template to be validated.
```
[the remark this answers is not in the document]
Comment (lahirumaramba) [reply src/index.d.ts]: Good catch! Thanks!

**Pull request merged**